File: 06_unique_values.js
// return true only if all values are unique, otherwise false
// using Array, object, string method lastIndexOf()
// no Set data structure

function unique(str){

    for (let i = 0; i < str.length; i++) {
        if(str.lastIndexOf(str[i])!==i){
            return false;
        };
    }
    return true;

    // object case
    // let values = {};
    // for (let letter of str) {
    //     if(values[letter]){
    //         return false;
    //     }
    //     values[letter] = "exists";
    // }
    // return true;


    // str case
    // let values = []
    // for (let letter of str) {
    //     if(values.indexOf(letter)!==-1){
    //         return false;
    //     }
    //     values.push(letter);
    // }
    // return true;
}

// true
console.log(unique("abced"));
// false
console.log(unique("abacdefb"));

File: 06_unique_values.py
# return true only if all values are unique. ohterwise false
# using Array, object, no set data structure

def unique(string):
    for i in range(len(string)):
        for j in range(len(string)):
            if i != j:
                if string[i] == string[j]:
                    return False
    return True


# en effective way
from collections import Counter

def effectiveUnique(string):
    counter = Counter(string)
    for values in counter.values():
        if values > 1:
            return False
    return True


print(unique("abced"))
print(unique("abacdefb"))
print("###############################")
print(effectiveUnique("abced"))
print(effectiveUnique("abacdefb"))
